app/test/unit/filter-changes-logic-test.ts

```diff
@@ -0,0 +1,478 @@
+          DiffSelectionType.All
+        ),
+      ]
+      const wd = createWorkingDirectory(files)
+      assert.equal(getNewFilesCount(wd), 2)
+    })
+
+    it('should count modified files correctly', () => {
+      const files = [
+        createTestFile('new.txt', AppFileStatusKind.New, DiffSelectionType.All),
+        createTestFile(
+          'modified1.txt',
+          AppFileStatusKind.Modified,
+          DiffSelectionType.All
+        ),
+        createTestFile(
+          'modified2.txt',
+          AppFileStatusKind.Modified,
+          DiffSelectionType.All
+        ),
+      ]
+      const wd = createWorkingDirectory(files)
+      assert.equal(getModifiedFilesCount(wd), 2)
+    })
+
+    it('should count deleted files correctly', () => {
+      const files = [
+        createTestFile(
+          'deleted1.txt',
+          AppFileStatusKind.Deleted,
+          DiffSelectionType.All
+        ),
+        createTestFile(
+          'modified.txt',
+          AppFileStatusKind.Modified,
+          DiffSelectionType.All
+        ),
+        createTestFile(
+          'deleted2.txt',
+          AppFileStatusKind.Deleted,
+          DiffSelectionType.All
+        ),
+      ]
+      const wd = createWorkingDirectory(files)
+      assert.equal(getDeletedFilesCount(wd), 2)
+    })
+
+    it('should count excluded files correctly', () => {
+      const files = [
+        createTestFile(
+          'included.txt',
+          AppFileStatusKind.Modified,
+          DiffSelectionType.All
+        ),
+        createTestFile(
+          'excluded1.txt',
+          AppFileStatusKind.Modified,
+          DiffSelectionType.None
+        ),
+        createTestFile(
+          'excluded2.txt',
+          AppFileStatusKind.Modified,
+          DiffSelectionType.None
+        ),
+      ]
+      const wd = createWorkingDirectory(files)
+      assert.equal(getExcludedFilesCount(wd), 2)
+    })
+
+    it('should count included files correctly', () => {
+      const files = [
+        createTestFile(
+          'included1.txt',
+          AppFileStatusKind.Modified,
+          DiffSelectionType.All
+        ),
+        createTestFile(
+          'included2.txt',
+          AppFileStatusKind.Modified,
+          DiffSelectionType.Partial
+        ),
+        createTestFile(
+          'excluded.txt',
+          AppFileStatusKind.Modified,
+          DiffSelectionType.None
+        ),
+      ]
+      const wd = createWorkingDirectory(files)
+      assert.equal(getIncludedFilesCount(wd), 2)
+    })
+  })
+
+  describe('getCheckBoxValueFromIncludeAll', () => {
+    it('should return On for true', () => {
+      assert.equal(getCheckBoxValueFromIncludeAll(true), CheckboxValue.On)
+    })
+
+    it('should return Off for false', () => {
+      assert.equal(getCheckBoxValueFromIncludeAll(false), CheckboxValue.Off)
+    })
+
+    it('should return Mixed for null', () => {
+      assert.equal(getCheckBoxValueFromIncludeAll(null), CheckboxValue.Mixed)
+    })
+  })
```

Comment:
Wouldn't bother testing or extracting `getCheckBoxValueFromIncludeAll` for testing. Method is simple and not likely to change.